### PR TITLE
GH-141 Fixing JournalLinkDecoder allowing to propagate 404 

### PR DESCRIPTION
**Pull request by francoisledroff**
## Description

* `IOErrorDecoder` polishing: only decode as upstream error when the response code is >=500
*  Fixing `JournalLinkDecode`r allowing to propagate `404` 
*  Polishing `JournalService`: adding a few error logs when the `JournalEntry` is null because of `404`
* a few other minor code clean up

## Related Issue

#141

### CODE REVIEW

File: core/src/main/java/com/adobe/aio/util/feign/FeignUtil.java [hunk 1]
```diff
@@ -39,10 +39,11 @@ private FeignUtil() {
    */
   public static Feign.Builder getBaseBuilder() {
     return Feign.builder()
+        .logger(new Slf4jLogger())
+        //.logLevel(Level.BASIC)
```
Comment (nicdard) [inline core/src/main/java/com/adobe/aio/util/feign/FeignUtil.java:43]: Should we pass a logger in here instead? As this is a util class, it may make sense to have a logger instance from the service using this FeignUtil class so that we can see from where the code was called, wdyt?

Comment (nicdard) [reply core/src/main/java/com/adobe/aio/util/feign/FeignUtil.java:43]: Also could you remove unused fields from this class?

File: core/src/main/java/com/adobe/aio/util/feign/IOErrorDecoder.java [hunk 2]
```diff
@@ -30,7 +30,7 @@ public class IOErrorDecoder implements ErrorDecoder {
   public Exception decode(String methodKey, Response response) {
     FeignException exception = FeignException.errorStatus(methodKey, response);
     logger.warn("Upstream response error ({},{})", response.status(), exception.contentUTF8());
-    return (response.status()>=400) ?
+    return (response.status()>=500) ?
```
Comment (nicdard) [inline core/src/main/java/com/adobe/aio/util/feign/IOErrorDecoder.java:33]: nit:
```suggestion
    return (response.status() >= 500) ?
```

File: events_test/src/main/java/com/adobe/aio/event/management/RegistrationServiceTester.java [hunk 3]
```diff
@@ -41,6 +41,10 @@ public RegistrationServiceTester() {
         .build();
   }
 
+  public RegistrationService getRegistrationService(){
+    return this.registrationService;
+  }
```
Comment (nicdard) [inline events_test/src/main/java/com/adobe/aio/event/management/RegistrationServiceTester.java:46]: where is this used?

Comment (francoisledroff) [reply events_test/src/main/java/com/adobe/aio/event/management/RegistrationServiceTester.java:46]: it's use in further tests we do internally within Adobe ;-)

File: events_journal/src/main/java/com/adobe/aio/event/journal/feign/FeignJournalService.java [hunk 4]
```diff
@@ -79,7 +89,11 @@ public JournalEntry get(String linkUrl) {
         .requestInterceptor(authInterceptor)
         .requestInterceptor(aioHeaderInterceptor)
         .target(JournalApi.class, linkUrl);
-    return linkJournalApi.get(this.imsOrgId);
+    JournalEntry journalEntry = linkJournalApi.get(this.imsOrgId);
+    if (journalEntry == null) {
+      logger.error("No/null Journal Entry, check your Journal link url: " + linkUrl);
+    }
```
[the remark this answers is not in the document]
Comment (francoisledroff) [reply events_journal/src/main/java/com/adobe/aio/event/journal/feign/FeignJournalService.java:95]: well here I also log a bit more ... using the `linkUrl`

File: core/src/main/java/com/adobe/aio/util/feign/FeignUtil.java (hunk 1 above)
Comment (francoisledroff) [reply core/src/main/java/com/adobe/aio/util/feign/FeignUtil.java:43]: as the logger yes I know it's sub-optimal, I created https://github.com/adobe/aio-lib-java/issues/146 for this.

Comment (nicdard) [reply core/src/main/java/com/adobe/aio/util/feign/FeignUtil.java:40]: Thanks!

File: events_test/src/main/java/com/adobe/aio/event/management/RegistrationServiceTester.java [hunk 5]
```diff
@@ -99,7 +103,6 @@ private static void assertUrl(String stringUrl) {
     try {
       Assert.assertNotNull(stringUrl);
       URL url = new URL(stringUrl);
-      Assert.assertEquals("https", url.getProtocol());
```
Comment (nicdard) [inline events_test/src/main/java/com/adobe/aio/event/management/RegistrationServiceTester.java]: Why is this removed?

**Review by nicdard — APPROVED** [review]
LGTM

**Comment by nicdard** [comment]
Side comment: If you do not want the unwanted committed changes to be in the history you could consider fixup the removals maybe, to have the history clean

File: events_test/src/main/java/com/adobe/aio/event/management/RegistrationServiceTester.java (hunk 5 above)
Comment (francoisledroff) [reply events_test/src/main/java/com/adobe/aio/event/management/RegistrationServiceTester.java]: to have the test running against a local non https url

Comment (nicdard) [reply events_test/src/main/java/com/adobe/aio/event/management/RegistrationServiceTester.java]: ok, got it
